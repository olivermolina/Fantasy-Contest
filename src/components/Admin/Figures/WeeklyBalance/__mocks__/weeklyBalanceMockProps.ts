export const weeklyBalanceMockData = [
  {
    agent: null,
    id: '1',
    username: 'jdoe',
    firstname: 'John',
    lastname: 'Doe',
    referral: '',
    isFirstDeposit: false,
    mondayBalance: 20,
    tuesdayBalance: 0,
    wednesdayBalance: -35,
    thursdayBalance: 0,
    fridayBalance: 0,
    saturdayBalance: 0,
    sundayBalance: 20,
    deposits: 50,
    withdrawals: 6,
    totalWeekBalance: 5,
    pendingTotal: 20,
    cashAmount: 0,
    withdrawable: 0,
    creditAmount: 0,
    totalBalance: 0,
    isActive: true,
  },
  {
    agent: null,
    id: '2',
    username: 'jwick',
    firstname: 'John',
    lastname: 'Wick',
    referral: '',
    isFirstDeposit: false,
    mondayBalance: 20,
    tuesdayBalance: 0,
    wednesdayBalance: -35,
    thursdayBalance: 0,
    fridayBalance: 0,
    saturdayBalance: 0,
    sundayBalance: 20,
    deposits: 50,
    withdrawals: 6,
    totalWeekBalance: 5,
    pendingTotal: 20,
    cashAmount: 0,
    withdrawable: 0,
    creditAmount: 0,
    totalBalance: 0,
    isActive: true,
  },
];
