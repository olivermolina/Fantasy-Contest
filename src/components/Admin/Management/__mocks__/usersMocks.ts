import { faker } from '@faker-js/faker';
import { UserType } from '@prisma/client';

export const usersMock = [
  {
    id: '1',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [
      {
        id: '1',
        userId: '1',
        code: 'cod1',
      },
      {
        id: '2',
        userId: '1',
        code: 'code2',
      },
    ],
    type: UserType.AGENT,
  },
  {
    id: '2',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [],
    type: UserType.PLAYER,
  },
  {
    id: '3',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [],
    type: UserType.PLAYER,
  },
  {
    id: '4',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [],
    type: UserType.PLAYER,
  },
];
