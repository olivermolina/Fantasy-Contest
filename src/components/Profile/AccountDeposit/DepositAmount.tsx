import React, {
  ChangeEvent,
  SetStateAction,
  useContext,
  useState,
} from 'react';
import { Button } from '~/components/Button';
import { AccountDepositContext } from './AccountDeposit';

const DepositInputAmount = () => {
  const { depositAmount, setDepositAmount, handleNext } = useContext(
    AccountDepositContext,
  );

  const [selected, setSelected] = useState<number | null>(null);

  const handleSelect = (
    amount: number,
    index: SetStateAction<number | null>,
  ) => {
    setDepositAmount?.(amount);
    setSelected(index);
  };

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setDepositAmount?.(Number(e.target.value));
    setSelected(null);
  };

  return (
    <div className="flex flex-col shadow-md rounded-md divide-y gap-2 bg-white">
      <div className="flex flex-col p-6 gap-2">
        <p className="font-bold text-lg">Account Deposit</p>
        <p className="text-sm">
          Select your deposit amount from the amounts shown below. or enter your
          own amount and click the &quot;Deposit&quot;.
        </p>
        <div className="grid grid-cols-2 justify-items-stretch gap-5 lg:gap-10 w-full mt-5">
          <div className="grid grid-cols-4 gap-2 justify-items-stretch">
            {[10, 50, 100, 250].map((amount, index) => (
              <button
                key={amount}
                className={`border border-gray-200 rounded-md font-bold text-md lg:text-lg shadow w-auto
                  ${selected === index ? 'border-4 border-blue-200' : ''}
                  `}
                onClick={() => handleSelect(amount, index)}
              >
                ${amount}
              </button>
            ))}
          </div>
          <div className="relative border border-gray-200 rounded-md bg-gray-200 ml-50 h-[62px]">
            <div className="flex absolute inset-y-0 left-0 items-center pointer-events-none bg-inherit p-2 rounded-l-md">
              <span className="font-bold text-lg">$</span>
            </div>
            <input
              type={'number'}
              className="block pl-11 pr-2 text-right appearance-none font-bold text-lg w-full h-full rounded-md bg-white focus:outline-gray-200"
              value={depositAmount}
              onChange={handleChange}
            />
          </div>
        </div>
      </div>
      <div className="flex flex-col p-6 gap-2">
        <Button variant="primary" width={'auto'} onClick={handleNext}>
          <p>Deposit</p>
        </Button>
        <p className="text-sm">
          Next: Select your deposit Method and save it to your secure wallet for
          future transactions.
        </p>
      </div>
    </div>
  );
};

export default DepositInputAmount;
