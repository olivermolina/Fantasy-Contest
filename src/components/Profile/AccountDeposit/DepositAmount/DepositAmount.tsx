import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
import { SubmitHandler, useForm } from 'react-hook-form';
import * as Yup from 'yup';
import { yupResolver } from '@hookform/resolvers/yup';
import { FormErrorText } from '~/components/Form/FormErrorText';
import classNames from 'classnames';
import { calculateReloadBonusAmount } from '~/utils/calculateReloadBonusAmount';

type Inputs = {
  depositAmount: number;
};

const InputValidationSchema = Yup.object().shape({
  depositAmount: Yup.number()
    .typeError('Please provide deposit amount')
    .min(10, 'Minimum deposit is $10'),
});

const styles = {
  defaultClasses:
    'block pl-11 pr-2 appearance-none font-bold text-lg w-full h-full rounded-md bg-[#1A487F] focus:outline-[#2F5F98]',
  errorClasses: 'border-2 border-red-500 focus:outline-red-500',
  defaultDepositButton:
    'transform transition duration-500 hover:scale-110 hover:border-4 hover:border-blue-300  border border-gray-200 rounded-md shadow w-auto py-2 min-h-[75px]',
};

interface DepositInputAmountProps {
  setDepositAmount: Dispatch<SetStateAction<number>>;
  handleNext: () => void;
  isFirstDeposit: boolean | undefined;
  depositAmount: number;
  maxMatchDeposit: number;
  depositAmountOptions: number[];
  /**
   * The free credit amount when reloading
   * @example 50
   **/
  reloadBonusAmount: number;
  /**
   * Type of reload bonus whe reloading
   * @example FLAT or PERCENTAGE
   **/
  reloadBonusType: string;
  /**
   * Boolean to show the bonus credit styles
   * @example true
   **/
  showBonusCreditStyles: boolean;
}

const DepositInputAmount = (props: DepositInputAmountProps) => {
  const {
    setDepositAmount,
    handleNext,
    isFirstDeposit,
    depositAmount,
    maxMatchDeposit,
    depositAmountOptions,
    reloadBonusAmount,
    reloadBonusType,
    showBonusCreditStyles,
  } = props;

  const {
    register,
    handleSubmit,
    formState: { errors, isValid },
    setValue,
    trigger,
    reset,
  } = useForm<Inputs>({
    resolver: yupResolver(InputValidationSchema),
  });

  const [selected, setSelected] = useState<number | null>(null);

  const handleSelect = async (
    amount: number,
    index: SetStateAction<number | null>,
  ) => {
    setDepositAmount?.(amount);
    setSelected(index);
    setValue('depositAmount', amount);
    await trigger('depositAmount');
  };

  const onSubmit: SubmitHandler<Inputs> = async (data) => {
    setDepositAmount?.(Number(data.depositAmount));
    handleNext?.();
  };

  useEffect(() => {
    if (depositAmount) {
      reset({ depositAmount });
    }
  }, [depositAmount]);

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <div className="flex flex-col shadow-md rounded-md divide-y gap-2 bg-primary w-full">
        <div className="flex flex-col p-6 gap-2">
          <p className="font-bold text-lg hidden lg:block">Account Deposit</p>
          <p className="text-sm">
            Select your deposit amount from the amounts shown below or enter
            your own amount and click the &quot;Deposit&quot; button.
          </p>
          <div className="flex flex-col gap-5 w-full mt-5 justify-center">
            <div className="grid grid-cols-2 md:grid-cols-3 gap-5 lg:gap-x-10 xl:gap-x-20 justify-items-stretch">
              {depositAmountOptions.map((amount, index) => (
                <button
                  key={amount}
                  className={classNames(styles.defaultDepositButton, {
                    'border-4 border-blue-300': selected === index,
                    'min-h-[105px]': showBonusCreditStyles,
                  })}
                  onClick={(e) => {
                    e.preventDefault();
                    handleSelect(amount, index);
                  }}
                >
                  <p className="font-bold text-2xl"> ${amount}</p>
                  <div className={'text-sm text-[#46B534] pt-2 font-semibold'}>
                    {isFirstDeposit && (
                      <>
                        <p>
                          +$
                          {amount > maxMatchDeposit ? maxMatchDeposit : amount}
                        </p>
                        <p>INSTANT Bonus</p>
                      </>
                    )}

                    {reloadBonusAmount > 0 && (
                      <>
                        <p>
                          +$
                          {calculateReloadBonusAmount(
                            amount,
                            reloadBonusAmount,
                            reloadBonusType,
                          ).toFixed(2)}
                        </p>
                        <p>Reload Bonus</p>
                      </>
                    )}
                  </div>
                </button>
              ))}
            </div>
            <p className={'text-center  font-bold'}>OR</p>
            <div className="relative border border-[#2F5F98] rounded-md bg-[#2F5F98] ml-50 h-[62px]">
              <div
                className={classNames(
                  'flex absolute inset-y-0 left-0 items-center pointer-events-none bg-inherit p-2 rounded-l-md',
                  errors?.depositAmount
                    ? 'border-2 border-red-500 focus:outline-red-500'
                    : '',
                )}
              >
                <span className="font-bold text-lg">$</span>
              </div>
              <input
                type="number"
                placeholder="Enter custom amount"
                className={classNames(
                  styles.defaultClasses,
                  errors?.depositAmount ? styles.errorClasses : '',
                )}
                defaultValue={''}
                {...register('depositAmount', {
                  onChange: (e) => {
                    setSelected(null);
                    setDepositAmount?.(Number(e.target.value));
                  },
                })}
              />
              <FormErrorText>{errors.depositAmount?.message}</FormErrorText>
            </div>
          </div>
        </div>
        <div className="flex flex-col p-6 gap-2">
          <button
            className="p-4 capitalize text-primary rounded font-bold w-auto h-auto bg-white disabled:opacity-10"
            type="submit"
            disabled={!isValid}
          >
            Deposit
          </button>
          <p className="text-sm text-lightText">
            Next: Select your deposit Method and save it to your secure wallet
            for future transactions.
          </p>
        </div>
      </div>
    </form>
  );
};

export default DepositInputAmount;
