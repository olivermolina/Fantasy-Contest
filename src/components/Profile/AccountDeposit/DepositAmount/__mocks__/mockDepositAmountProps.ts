export const mockDepositAmountProps = {
  setDepositAmount: () => {
    alert('Set Deposit amount');
  },
  handleNext: () => {
    alert('handleNext');
  },
  depositAmount: 10,
  isFirstDeposit: false,
  maxMatchDeposit: 50,
  depositAmountOptions: [10, 25, 50, 75, 100, 125],
  reloadBonusAmount: 0,
  reloadBonusType: 'FLAT',
  showBonusCreditStyles: false,
  minDepositAmount: 10,
};
