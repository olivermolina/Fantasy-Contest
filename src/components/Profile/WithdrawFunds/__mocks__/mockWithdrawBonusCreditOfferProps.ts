export const mockWithdrawBonusCreditOfferProps = {
  handleSubmit: () => console.log('submit'),
  handleNext: () => console.log('next'),
  open: true,
  setOpen: () => console.log('set open'),
  maxRetentionBonus: 500,
  retentionBonusMatchMultiplier: 2,
  cashBalance: 100,
};
