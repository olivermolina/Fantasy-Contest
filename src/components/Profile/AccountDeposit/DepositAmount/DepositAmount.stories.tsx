import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import Deposit from '.';
import { mockDepositAmountProps } from './__mocks__/mockDepositAmountProps';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Lockspread/Profile',
  component: Deposit,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
    isFirstDeposit: {
      options: [true, false],
      control: { type: 'radio' },
    },
  },
} as ComponentMeta<typeof Deposit>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Deposit> = (args) => (
  <Deposit {...args} />
);

export const DepositAmount = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
DepositAmount.args = mockDepositAmountProps;
