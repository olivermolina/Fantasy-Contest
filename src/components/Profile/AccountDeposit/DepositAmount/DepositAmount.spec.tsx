import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import DepositAmount from './DepositAmount';
import { mockDepositAmountProps } from './__mocks__/mockDepositAmountProps';

describe('DepositAmount', () => {
  it('should render correctly', () => {
    const { container } = render(<DepositAmount {...mockDepositAmountProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should not render first deposit match INSTANT Bonus or Reload Bonus ', () => {
    render(<DepositAmount {...mockDepositAmountProps} />);

    expect(screen.queryAllByText('INSTANT Bonus')).toHaveLength(0);
    expect(screen.queryAllByText('Reload Bonus')).toHaveLength(0);
  });

  it('should render first deposit match INSTANT Bonus', () => {
    render(<DepositAmount {...mockDepositAmountProps} isFirstDeposit={true} />);

    expect(screen.queryAllByText('INSTANT Bonus')).toHaveLength(6);
  });

  it('should render Reload Bonus text when AppSetting.RELOAD_BONUS_AMOUNT is greater than 0', () => {
    render(
      <DepositAmount {...mockDepositAmountProps} reloadBonusAmount={50} />,
    );

    expect(screen.queryAllByText('Reload Bonus')).toHaveLength(6);
  });
});
