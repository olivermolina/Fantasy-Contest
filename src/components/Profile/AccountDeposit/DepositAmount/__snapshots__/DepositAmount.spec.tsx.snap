// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositAmount should render correctly 1`] = `
<div>
  <form>
    <div
      class="flex flex-col shadow-md rounded-md divide-y gap-2 bg-primary w-full"
    >
      <div
        class="flex flex-col p-6 gap-2"
      >
        <p
          class="font-bold text-lg hidden lg:block"
        >
          Account Deposit
        </p>
        <p
          class="text-sm"
        >
          Select your deposit amount from the amounts shown below or enter your own amount and click the "Deposit" button.
        </p>
        <div
          class="flex flex-col gap-5 w-full mt-5 justify-center"
        >
          <div
            class="grid grid-cols-2 md:grid-cols-3 gap-5 lg:gap-x-10 xl:gap-x-20 justify-items-stretch"
          >
            <button
              class="transform transition duration-500 hover:scale-110 hover:border-4 hover:border-blue-300  border border-gray-200 rounded-md shadow w-auto py-2 min-h-[75px]"
            >
              <p
                class="font-bold text-2xl"
              >
                 $
                10
              </p>
              <div
                class="text-sm text-[#46B534] pt-2 font-semibold"
              />
            </button>
            <button
              class="transform transition duration-500 hover:scale-110 hover:border-4 hover:border-blue-300  border border-gray-200 rounded-md shadow w-auto py-2 min-h-[75px]"
            >
              <p
                class="font-bold text-2xl"
              >
                 $
                25
              </p>
              <div
                class="text-sm text-[#46B534] pt-2 font-semibold"
              />
            </button>
            <button
              class="transform transition duration-500 hover:scale-110 hover:border-4 hover:border-blue-300  border border-gray-200 rounded-md shadow w-auto py-2 min-h-[75px]"
            >
              <p
                class="font-bold text-2xl"
              >
                 $
                50
              </p>
              <div
                class="text-sm text-[#46B534] pt-2 font-semibold"
              />
            </button>
            <button
              class="transform transition duration-500 hover:scale-110 hover:border-4 hover:border-blue-300  border border-gray-200 rounded-md shadow w-auto py-2 min-h-[75px]"
            >
              <p
                class="font-bold text-2xl"
              >
                 $
                75
              </p>
              <div
                class="text-sm text-[#46B534] pt-2 font-semibold"
              />
            </button>
            <button
              class="transform transition duration-500 hover:scale-110 hover:border-4 hover:border-blue-300  border border-gray-200 rounded-md shadow w-auto py-2 min-h-[75px]"
            >
              <p
                class="font-bold text-2xl"
              >
                 $
                100
              </p>
              <div
                class="text-sm text-[#46B534] pt-2 font-semibold"
              />
            </button>
            <button
              class="transform transition duration-500 hover:scale-110 hover:border-4 hover:border-blue-300  border border-gray-200 rounded-md shadow w-auto py-2 min-h-[75px]"
            >
              <p
                class="font-bold text-2xl"
              >
                 $
                125
              </p>
              <div
                class="text-sm text-[#46B534] pt-2 font-semibold"
              />
            </button>
          </div>
          <p
            class="text-center  font-bold"
          >
            OR
          </p>
          <div
            class="relative border border-[#2F5F98] rounded-md bg-[#2F5F98] ml-50 h-[62px]"
          >
            <div
              class="flex absolute inset-y-0 left-0 items-center pointer-events-none bg-inherit p-2 rounded-l-md"
            >
              <span
                class="font-bold text-lg"
              >
                $
              </span>
            </div>
            <input
              class="block pl-11 pr-2 appearance-none font-bold text-lg w-full h-full rounded-md bg-[#1A487F] focus:outline-[#2F5F98]"
              name="depositAmount"
              placeholder="Enter custom amount"
              type="number"
              value=""
            />
            <span
              class="text-xs text-red-500 ml-3"
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col p-6 gap-2"
      >
        <button
          class="p-4 capitalize text-primary rounded font-bold w-auto h-auto bg-white disabled:opacity-10"
          disabled=""
          type="submit"
        >
          Deposit
        </button>
        <p
          class="text-sm text-lightText"
        >
          Next: Select your deposit Method and save it to your secure wallet for future transactions.
        </p>
      </div>
    </div>
  </form>
</div>
`;
